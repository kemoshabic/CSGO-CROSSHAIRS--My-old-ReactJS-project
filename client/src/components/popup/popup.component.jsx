import React from 'react'
import { connect } from 'react-redux'
import { createStructuredSelector } from 'reselect'

import { selectExistingError } from '../../redux/user/user.selectors'
import { togglePopUp } from "../../redux/user/user.actions"

import Error from '../../assets/error.png'
import CloseButton from '../../assets/x-button.png'

import './popup.styles.scss';

export const PopUp = ({ togglePopUp, error }) => {
  return (
   <div className="error">
     <div className="error__box-1">
       <img src={Error} alt="error button" className="error__img"/>
       <img src={CloseButton} alt="close button" onClick={togglePopUp} className="error__close"/>
    </div>
    <div className="error__box-2">
       <p className="error__main-text">Oh snap!</p>
       <p className="error__text">{error.message}</p>
    </div>
   </div>
  );
 }

const mapStateToProps = createStructuredSelector({
  error: selectExistingError,
})

const mapDispatchToProps = (dispatch) => ({
  togglePopUp: () => dispatch(togglePopUp())
})

export default connect(mapStateToProps, mapDispatchToProps)(PopUp)