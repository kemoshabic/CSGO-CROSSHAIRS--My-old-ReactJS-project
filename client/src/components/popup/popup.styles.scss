.error {
  width: 60rem;
  height: 35rem;
  background-color: #d84242;
  display: flex;
  flex-direction: column;
  margin-left: 20vw;
  @media only screen and (max-width: 81.25em) {
    // 1300px
    margin-left: 15vw;
  }

  @media only screen and (max-width: 57.5em) {
    // 920px
    margin-left: 25vw;
  }
  @media only screen and (max-width: 45.5em) {
    // 920px
    margin-left: 15vw;
  }

  @media only screen and (max-width: 68.75em) {
    // 1100px
    width: 40rem;
    height: 20rem;
  }
  &__img {
    width: 5rem;
  }

  &__box-1 {
    height: 17rem;
    width: 60rem;
    background-color: #ffffff;
    display: flex;
    justify-content: center;
    align-items: center;
    position: relative;
    @media only screen and (max-width: 68.75em) {
      // 1100px
      width: 40rem;
      height: 10rem;
    }
  }
  &__close {
    position: absolute;
    top: 0;
    right: -1rem;
    cursor: pointer;
  }
  &__box-2 {
    height: 17.5rem;
    width: 60rem;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    @media only screen and (max-width: 68.75em) {
      // 1100px
      width: 40rem;
      height: 10rem;
    }
  }
  &__main-text {
    font-size: 2.8rem;
    @media only screen and (max-width: 68.75em) {
      font-size: 1.8rem;
    }
  }
  &__text {
    margin-top: -0.2rem;
    font-size: 2.4rem;
    color: #0c0e25;
    text-align: center;
    @media only screen and (max-width: 68.75em) {
      font-size: 1.4rem;
    }
  }
}
