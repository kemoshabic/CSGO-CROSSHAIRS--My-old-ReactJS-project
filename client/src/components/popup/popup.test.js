import React from "react";
import { shallow } from "enzyme";

import { PopUp } from "./popup.component"

describe("PopUp component", () => {
  let wrapper;
  let mockTogglePopUp;

  beforeEach(() => {
    mockTogglePopUp = jest.fn();

    const mockProps = {
      error: {
        message: "error logging in"
      },
      togglePopUp: mockTogglePopUp,
    };

    wrapper = shallow(<PopUp {...mockProps} />);
  });

  it('should render PopUp component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should return errorText length greater than 0", () => {
    const errorText = wrapper.find('.error__text').text();
    expect(errorText.length).toBeGreaterThan(0)
  })

  it("should return no error message if error message is not defined", () => {
    const mockProps = {
      error: {
        message: ""
      },
      togglePopUp: mockTogglePopUp,
    };

    const newWrapper = shallow(<PopUp {...mockProps} />);
    const errorText = newWrapper.find('.error__text').text();
    expect(errorText).toEqual('');
  })
  it("should call togglePopUp", () => {
    wrapper.find('.error__close').simulate('click');
    expect(mockTogglePopUp).toHaveBeenCalled();
  })
});
