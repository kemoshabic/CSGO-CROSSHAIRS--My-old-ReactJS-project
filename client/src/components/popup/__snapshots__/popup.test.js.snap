// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopUp component should render PopUp component 1`] = `ShallowWrapper {}`;
